hs_core/views/__init__.py

```diff
@@ -96,6 +99,43 @@ def verify(request, *args, **kwargs):
     return HttpResponseRedirect('/')
 
 
+@login_required
+def get_tasks_by_user(request):
+    task_list = get_all_tasks(request.user.username)
+    return JsonResponse({'tasks': task_list})
+
+
+@login_required
+def get_task(request, task_id):
+    task_dict = get_task_by_id(task_id)
+    return JsonResponse(task_dict)
+
+
+@login_required
```

Comment:
Shouldn't this method also check the logged in user against the user who created the task_id here and the methods below?

Reply:
A possible approach is to include the user on the TaskNotification model and include the user in the filter

Reply:
If I understand your question correctly, you meant that we need to handle the case where the logged in user can see the tasks created by other users? Since the task notification app allows the user to abort a task, I don't think it is a good idea to allow the login user to see any tasks created by other users, but instead the login user can only see all tasks created by the login user. I do see an issue that is not handled already, specifically when multiple owners are deleting the same resource at about the same time, similarly to when multiple users are downloading the same resource bag. I think I should be able to add the logic similar to bag download here. Specifically, instead of creating a resource deletion task regardless, it needs to check whether there is already a task already running that tries to delete the same resource, and if yes, it just needs to wait for that task to complete rather than creating a new task. I can look into it and add this logic. Let me know if this is not what you meant in your comment above.

Reply:
I'm just thinking about only allowing a user who created a task to do something like abort a task that the person created.  The way I read this method, as long as I'm logged in and have a task_id I can delete any task.  Instead I think it makes sense to authenticate the user aborting the task actually has permission to abort the specific task they're requesting.

The issue you highlight is an important one and probably can be implemented by lifting the logic used for downloading a bag and resuming the download while it's running in celery (which you probably already know).

Reply:
Check out my latest commit that should address your comment as well as adding the logic to check whether there is an existing task already for deleting the same resource.